src/lava/lib/optimization/solvers/bayesian/models.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# See: https://spdx.org/licenses/
+
+import cv2
+import matplotlib.pylab as plt
+import numpy as np
+import os
+from scipy.optimize import OptimizeResult
+from skopt import Optimizer, Space
+import skopt.plots as skplots
+from skopt.space import Categorical, Integer, Real
+from typing import Union
+
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+
+from lava.lib.optimization.solvers.bayesian.processes import (
+    BayesianOptimizer
+)
+
+
+@implements(proc=BayesianOptimizer, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyBayesianOptimizerModel(PyLoihiProcessModel):
+    """
+    A Python-based implementation of the Bayesian Optimizer processes. For
+    more information, please refer to bayesian/processes.py.
+    """
+    results_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, np.float64)
+    next_point_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, np.float64)
+
+    acq_func_config = LavaPyType(np.ndarray, np.ndarray)
+    acq_opt_config = LavaPyType(np.ndarray, np.ndarray)
+    search_space = LavaPyType(np.ndarray, np.ndarray)
+    enable_plotting = LavaPyType(np.ndarray, np.ndarray)
+    est_config = LavaPyType(np.ndarray, np.ndarray)
+    ip_gen_config = LavaPyType(np.ndarray, np.ndarray)
+    log_dir = LavaPyType(np.ndarray, np.ndarray)
+    num_ips = LavaPyType(int, int)
+    num_objectives = LavaPyType(int, int)
+    seed = LavaPyType(int, int)
+
+    initialized = LavaPyType(bool, bool)
+    num_iterations = LavaPyType(int, int)
+    frame_log = LavaPyType(np.ndarray, np.ndarray)
+
+    def run_spk(self) -> None:
+        """tick the model forward by one time-step"""
+
+        if self.initialized:
+            # receive a result vector from the black-box function
+            result_vec: np.ndarray = self.results_in.recv()
+
+            self.opt_result: OptimizeResult = self.process_result_vector(
+                result_vec
+            )
+
+            if self.enable_plotting[0]:
```

Comment:
I think the decision to factor the guts of the optimizer (above) into a supporting module was great design. I would suggest applying the same logic to the plotting code. Details of image formatting (and as already mentioned, which package to use for plotting) can be abstracted away to avoid clutter.